diag_manager/fms_diag_yaml_object.F90

```diff
@@ -67,38 +59,38 @@ module fms_diag_yaml_object_mod
      character (len=:), allocatable :: file_unlimdim !< The name of the unlimited dimension
      logical :: file_write
-     character (len=:), allocatable :: string_file_write !< false if the user doesn’t want the file to be 
+     character (len=:), allocatable :: string_file_write !< false if the user doesn’t want the file to be
                                                !! created (default is true).
      character (len=:), allocatable :: file_realm !< The modeling realm that the variables come from
-     real :: file_region (NUM_REGION_ARRAY) !< Bounds of the regional section to capture
+     type(sub_region_type) :: file_sub_region !< type containing info about the subregion, if any
      integer :: file_new_file_freq !< Frequency for closing the existing file
-     character (len=:), allocatable :: file_new_file_freq_units !< Time units for creating a new file. 
+     character (len=:), allocatable :: file_new_file_freq_units !< Time units for creating a new file.
                                                         !! Required if “new_file_freq” used
-     integer :: file_start_time !< Time to start the file for the first time. Requires “new_file_freq”
-     integer :: file_duration !< How long the file should receive data after start time 
-                                      !! in “file_duration_units”.  This optional field can only 
+     character (len=:), allocatable :: file_start_time !< Time to start the file for the first time. Requires
+                                                       !! “new_file_freq”
+     integer :: file_duration !< How long the file should receive data after start time
+                                      !! in “file_duration_units”.  This optional field can only
                                       !! be used if the start_time field is present.  If this field
                                       !! is absent, then the file duration will be equal to the
-                                      !! frequency for creating new files.  
+                                      !! frequency for creating new files.
                                       !! NOTE: The file_duration_units field must also be present if
                                       !! this field is present.
     character (len=:), allocatable :: file_duration_units !< The file duration units
-    character (len=:), dimension(:), allocatable :: file_varlist !< An array of variable names
+    character (len=MAX_STR_LEN), dimension(:), allocatable :: file_varlist !< An array of variable names
                                                              !! within a file
-    character (len=:), dimension(:,:), allocatable :: file_global_meta !< Array of key(dim=1) 
-                                                        !! and values(dim=2) to be added as global 
+    character (len=MAX_STR_LEN), dimension(:,:), allocatable :: file_global_meta !< Array of key(dim=1)
```

Comment:
I think you should add a comment about why `len=MAX_STR_LEN` and not `:` for someone else who will have to deal with this in the future.